script/vm/compiler.lua

```diff
@@ -101,6 +101,26 @@ local function searchFieldByLocalID(source, key, pushResult)
     if not fields then
         return
     end
+
+    --Exact classes do not allow injected fields
+    if source.type ~= 'variable' and source.bindDocs then
+        ---@cast source parser.object
+        local uri = guide.getUri(source)
+        for _, src in ipairs(fields) do
+            if src.type == "setfield" then
+                local class = vm.getDefinedClass(uri, source)
+                if class then
+                    for _, doc in ipairs(class:getSets(uri)) do
+                        if vm.docHasAttr(doc, 'exact') then
+                            return
+                        end
+                    end
+                end
+            end
+        end
```

Comment:
The forloop here seems can early break 🤔 
1. `local class = vm.getDefinedClass(uri, source)` doesn't depend on `src`, so it is a constant in this loop
2. Now whether `class` is nil or not, it is unrelated to outer loop, and checking the whole logic **once** is enough:
    - if the class has `exact` attribute => early `return`
    - if the class is `nil` or has no `exact` attribute => early `break`

```lua
        for _, src in ipairs(fields) do
            if src.type == "setfield" then
                local class = vm.getDefinedClass(uri, source)
                if class then
                    for _, doc in ipairs(class:getSets(uri)) do
                        if vm.docHasAttr(doc, 'exact') then
                            return
                        end
                    end
                end
                break  <-- the break here, because the above logic only need to check once
            end
        end
```